python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -626,6 +653,43 @@ def _create_initial_sensor_cursors_from_raw_cursor(
 
         return result
 
+    def _rename_auto_materialize_states(
+        self,
+        instance: DagsterInstance,
+        all_auto_materialize_states: Mapping[str, InstigatorState],
+        sensors_and_repos: Sequence[Tuple[ExternalSensor, ExternalRepository]],
+    ) -> Mapping[str, InstigatorState]:
+        """Searches for sensors named `default_auto_materialize_sensor` and migrates their state
+        to a sensor in the same repository named `default_automation_condition_sensor`.
+        """
+        result = dict(all_auto_materialize_states)
+
+        for instigator_state in all_auto_materialize_states.values():
+            instigator_state_name = instigator_state.origin.instigator_name
+            instigator_state_repository_origin = instigator_state.origin.repository_origin
+            # only migrate instigators with the name "default_auto_materialize_sensor"
+            if instigator_state_name != "default_auto_materialize_sensor":
+                continue
+            for sensor, repo in sensors_and_repos:
+                # find the corresponding sensor with the updated name
+                if (
+                    sensor.name == "default_automation_condition_sensor"
+                    and repo.get_external_origin() == instigator_state_repository_origin
+                ):
+                    new_auto_materialize_state = InstigatorState(
+                        sensor.get_external_origin(),
+                        InstigatorType.SENSOR,
+                        instigator_state.status,
+                        instigator_state.instigator_data,
+                    )
+                    result[sensor.selector_id] = new_auto_materialize_state
```

Comment:
if you took out sensors_and_repos this would then become

result[new_sensor_origin.get_selector().get_id()]